lib/api/api.js

```diff
@@ -643,15 +643,8 @@ api.prototype.sendTransfer = function(seed, depth, minWeightMagnitude, transfers
 */
 api.prototype.promoteTransaction = function(tail, depth, minWeightMagnitude, transfer, params, callback) {
     var self = this;
-    if (!inputValidator.isHash(tail)) {
-        return callback(errors.invalidTrytes());
-    }
 
     self.isPromotable(tail).then(function (isPromotable) {
-      if (!isPromotable) {
```

Comment:
This is required for stopping promotion, if tail is inconsistent. Shouldn't be removed.